pkg/types/gcp/platform.go

```diff
@@ -65,6 +75,13 @@ type Platform struct {
 	// There must be only one ServiceEndpoint for a service.
 	// +optional
 	ServiceEndpoints []configv1.GCPServiceEndpoint `json:"serviceEndpoints,omitempty"`
+
+	// PrivateZone contains the information for a private DNS zone. The Private DNS Zone can
+	// only be supplied during Shared VPC (XPN) installs. The PrivateZone can exist or be
+	// created in a second service project; a project other than the one matching projectID
+	// or networkProjectID.
+	// +optional
+	PrivateZone *DNSZone `json:"privateZone,omitempty"`
```

Comment:
It would be helpful if the variable name included `dns`.

Right now I think the install config stanza looks something like:

```yaml
gcp:
  privateZone:
    projectID: dns-project
    zone: zone-name
```

Perhaps just nesting it in a dns object (and incorporating zone-> name suggestion)

```yaml
gcp:
  dns:
    privateZone:
      projectID: dns-project
      zone: zone-name
```